File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        map = {}

        for i, n in enumerate(numbers):
            diff = target - n
            if diff in map and map[diff] != i:
                return [map[diff] + 1, i + 1]
            map[n] = i
        return []

File: Data Structures & Algorithms/two-integer-sum/submission-4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> map;
        for(int i = 0; i < nums.size(); i++) {
            map[nums[i]] = i; // {value, index}
        }
        for(int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            if(map.contains(diff) && map[diff] != i) {
                if(map[diff] > i) return {i, map[diff]};
                else return {map[diff], i};
            }
        }
        return {};
    }
};
